### ISSUE TITLE
Operator 1.6 : Remove "AllowPrivilegeEscalation" or decouple "AllowPrivilegeEscalation" from "Privileged"

### ISSUE THREAD

**Issue by neelasha-09** [zalando/postgres-operator#1403]
Hi Team,

While using OPR v1.6.1, we get the error  `seteuid: Operation not permitted` due to cron binary unable to run.

https://github.com/zalando/spilo/issues/562

We compared the old (1.5.0)and new operator (1.6.1) and observed the error is due to the `allowPrivilegeEscalation: false` under `securityContext`.

We see required  either to remove the  `allowPrivilegeEscalation` ( as in 1.5.0 ) or to decouple the `allowPrivilegeEscalation` from `Privileged` by having 2 different parameters:

Reference links below for the code:

**Old OPR: 1.5.0 - Working**
https://github.com/zalando/postgres-operator/blob/a9b677c957be09de5e02f3b0eec4221c551eb2ca/pkg/cluster/k8sres.go#L456

**New OPR: 1.6.1 - cron error**
https://github.com/zalando/postgres-operator/blob/ff8143770c9ca5b2e7b36d1d29d8c914e424b9c9/pkg/cluster/k8sres.go#L469

In conclusion, at least for OpenShift  we need the parameter  `allowPrivilegeEscalation` to be set as `true` and  `Privileged` to be set as `false`

**Comment by neelasha-09** [comment]
@FxKu : Could you please review the PR  #1412
